src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4842,13 +4842,39 @@ public PreprocessingSymbolInfo GetPreprocessingSymbolInfo(IdentifierNameSyntax n
         {
             CheckSyntaxNode(node);
 
-            if (node.Ancestors().Any(n => SyntaxFacts.IsPreprocessorDirective(n.Kind())))
+            if (node.Ancestors().Any(n => isPreprocessorDirectiveAcceptingPreprocessingSymbols(n.Kind())))
             {
-                bool isDefined = this.SyntaxTree.IsPreprocessorSymbolDefined(node.Identifier.ValueText, node.Identifier.SpanStart);
-                return new PreprocessingSymbolInfo(new Symbols.PublicModel.PreprocessingSymbol(node.Identifier.ValueText), isDefined);
+                return CreatePreprocessingSymbolInfo(node.Identifier);
             }
 
             return PreprocessingSymbolInfo.None;
+
+            static bool isPreprocessorDirectiveAcceptingPreprocessingSymbols(SyntaxKind kind)
+            {
+                switch (kind)
+                {
+                    case SyntaxKind.IfDirectiveTrivia:
+                    case SyntaxKind.ElifDirectiveTrivia:
+                    case SyntaxKind.DefineDirectiveTrivia:
+                    case SyntaxKind.UndefDirectiveTrivia:
+                        return true;
+                    default:
+                        return false;
+                }
+            }
```

Comment:
These changes to `GetPreprocessingSymbolInfo`, which were made to avoid the identifier references in `#pragma warning` directives, overlap with the changes in #72910. I think it's better to remove the changes here and let that other PR handle the case.